tests/cli/test_cli.py

```diff
@@ -1771,6 +1775,29 @@ def test_run(ctx):
     )
     _run_command([cl, 'uedit', 'codalab', '--time-quota', ctx.time_quota])  # reset time quota
 
+    # Test that bundle fails when run without sufficient disk quota
+    # Note: We add 1MB to the disk quota since that's the minimum disk allowed to be requested
+    # by a container.
+    # We then create a file with size 1MB + 10 bytes, since that should now violate the disk quota.
+    disk_used = _run_command([cl, 'uinfo', 'codalab', '-f', 'disk_used'])
+    _run_command([cl, 'uedit', 'codalab', '--disk-quota', f'{int(disk_used) + 1000000}'])
+    uuid = _run_command(
+        [cl, 'run', f'head -c {1000010} /dev/zero > test.txt; sleep 100000',],
+        request_disk=None,
+        request_memory=None,
+    )
+    wait_until_state(
+        uuid, State.KILLED, timeout_seconds=500, exclude_final_states={State.FAILED}
+    )  # exclude State.FAILED because upon the initial upload attempt, the bundle state is set to FAILED
```

Comment:
What does "exclude State.FAILED" mean?

Reply:
It just means that wait_until_state should keep running even if the bundle reaches state State.FAILED. In this case, because of how the bundle is updated, that isn't actually a final state (it is updated to State.KILLED after being updated to State.FAILED as detailed in my reply to your comment right above this).